jobs/uaa/spec

```diff
@@ -832,6 +832,9 @@ properties:
     default: true
   uaa.database.case_insensitive:
     description: "Set to true if you don't want to be using LOWER() SQL functions in search queries/filters, because you know that your DB is case insensitive. If this property is null, then it will be set to true if the UAA DB is MySQL and false otherwise, but even on MySQL you can override it by setting it explicitly to false"
+  uaa.database.test_while_idle:
+    description: "If true, connections will be validated by the idle connection evictor (if any). If the validation fails, the connection is destroyed and removed from the pool."
```

Comment:
I think that this description is confusing. This is probably why I asked the previous question about the name.

Related to my comment about renaming the variable, could the description just say something like
> If true, broken connections will be evicted from the connection pool. Else, they will just stay there forever.

Which really brings back the question: why would anyone want that field set to false?

Reply:
@bruce-ricard I synced with @tack-sap. We cannot answer why somebody would want to set it false. But we guess the feature requires some background work (connection testing) that might not be wanted in some cases.

Regarding the description, we took it over from Apache
"If true connections will be validated by the idle connection evictor (if any). If the validation fails, the connection is destroyed and removed from the pool"
http://tomee.apache.org/latest/docs/datasource-config.html

Spring has a variation of that:
"The indication of whether objects will be validated by the idle object evictor (if any). If an object fails to validate, it will be dropped from the pool."
https://docs.spring.io/spring-ldap/site/reference/html/pooling.html

Note that also there the default is false. In general we are open to adapt the description if wanted.

Reply:
I think that there's something I don't understand, and I would like to understand it at some point. Maybe we can go ahead and merge the PR since it looks similar to what Spring and Apache do, but I would like to understand what is going on here. I really would like to answer the question: why does this seem to be something commonly configurable? Why would anyone want to configure it not to evict broken connections from the pool?

My current position is that I don't like the name "test_while_idle", because it seems too low level to me. As I have mentioned in a previous reply, I would like it to be higher level. IMHO what matters to the operator is that the connections get evicted, not that they get tested, that's just an implementation detail.